debezium-server-kafka/src/main/java/io/debezium/server/kafka/KafkaChangeConsumer.java

```diff
@@ -81,31 +88,35 @@ void stop() {
     public void handleBatch(final List<ChangeEvent<Object, Object>> records,
                             final RecordCommitter<ChangeEvent<Object, Object>> committer)
             throws InterruptedException {
-        final CountDownLatch latch = new CountDownLatch(records.size());
         for (ChangeEvent<Object, Object> record : records) {
             try {
                 LOGGER.trace("Received event '{}'", record);
                 Headers headers = convertKafkaHeaders(record);
 
                 String topicName = streamNameMapper.map(record.destination());
-                producer.send(new ProducerRecord<>(topicName, null, null, record.key(), record.value(), headers), (metadata, exception) -> {
-                    if (exception != null) {
-                        LOGGER.error("Failed to send record to {}:", topicName, exception);
-                        throw new DebeziumException(exception);
-                    }
-                    else {
-                        LOGGER.trace("Sent message with offset: {}", metadata.offset());
-                        latch.countDown();
-                    }
-                });
+                Future<RecordMetadata> recordMetadataFuture = producer.send(new ProducerRecord<>(topicName, null, null, record.key(), record.value(), headers),
+                        (metadata, exception) -> {
+                            if (exception != null) {
+                                LOGGER.error("Failed to send record to {}:", topicName, exception);
+                                throw new DebeziumException(exception);
+                            }
+                            else {
+                                LOGGER.trace("Sent message with offset: {}", metadata.offset());
+                            }
+                        });
+                if (waitMessageDeliveryTimeout == 0) {
+                    recordMetadataFuture.get();
+                }
+                else {
+                    recordMetadataFuture.get(waitMessageDeliveryTimeout, TimeUnit.MILLISECONDS);
```

Comment:
Why do we have our own timeout handling here instead of using `delivery.timeout.ms` option for the KafkaProducer / ProducerConfig?
Are you aware that the KafkaProducer has this config option and it is set to 120_000 ms = 2 minutes?
You default to 30_000 / 30 seconds. That can lead to more errors than they really happen because you only wait for 30 seconds while the producer might return after 2 mins, when I understand the code correctly?!
Am I wrong here? @jpechane ?

Reply:
This is good point. The code still should stay in place as a defensive measure but the default should be changed to correspond to Kafka default (or slightly more).

Reply:
I can open a PR really quick and set the default value to match the `delivery.timeout.ms` config.
What I've noticed in our case is that even with this timeout, the producer might retry indefinitely to push some records to Kafka since default values are very permisive and we ended up with WAL increasing but Debezium not crashing. I feel like this config will allow us to crash hard the server and rely on a full restart for a recovery, maybe...

Reply:
This `recordMetadataFuture.get`  for each record kills the throughput of the Kafka Producer.

Reply:
We see lots of ASYNC_NETWORK_IO waits in SQL Server with this in place (https://www.sqlshack.com/reducing-sql-server-async_network_io-wait-type/). This did not happen in the previous version and it indicates that the SQL Server client, debezium server in this case, is slow in processing query results.